src/components/Login.vue

```diff
@@ -212,6 +175,22 @@ export default {
           this.errorMsg = err.message;
         });
     },
+    checkEmailAccess(snapshot, user) {
+      let isAlphaEmail = false;
+      snapshot.forEach(doc => {
+        if (doc.data().email === user.user.email) {
+          isAlphaEmail = true;
+        }
+      });
+      this.performingRequest = false;
+      if (!isAlphaEmail) {
+        fb.auth.signOut();
+        alert('Sorry, but you do not have alpha access.\nPlease sign up below for email updates on when the platform is available and for a chance to test the platform early.');
+        return false;
+      }
+
+      return true;
+    },
```

Comment:
This looks bad. You are implicitly exposing the alpha whitelist email to the entire world, since you need to loop over the entire snapshot.

Reply:
My suggestion:
1. Write firebase security rules so that user can only access his/her own whitelist document
2. Try to access that document
3. If caught insufficient permission error, then we know that the user is not whitelisted.